app/src/main/java/uk/gov/onelogin/developer/tabs/auth/AuthTabScreenViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.onelogin.developer.tabs.auth
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+import uk.gov.onelogin.network.usecase.HelloWorldApiCall
+import uk.gov.onelogin.repositiories.TokenRepository
+
+@HiltViewModel
+class AuthTabScreenViewModel @Inject constructor(
+    private val helloWorldApiCall: HelloWorldApiCall,
+    private val tokenRepository: TokenRepository
+) : ViewModel() {
+    private val _happyHelloWorldResponse = mutableStateOf("")
+    val happyHelloWorldResponse: State<String>
+        get() = _happyHelloWorldResponse
+
+    private val _happyCallLoading = mutableStateOf(false)
+    val happyCallLoading: State<Boolean>
+        get() = _happyCallLoading
+
+    private val _sadHelloWorldResponse = mutableStateOf("")
+    val sadHelloWorldResponse: State<String>
+        get() = _sadHelloWorldResponse
+
+    private val _sadCallLoading = mutableStateOf(false)
+    val sadCallLoading: State<Boolean>
+        get() = _sadCallLoading
+
+    fun makeHappyHelloWorldCall() {
+        _happyCallLoading.value = true
+        viewModelScope.launch {
+            _happyHelloWorldResponse.value = helloWorldApiCall()
+            _happyCallLoading.value = false
+        }
+    }
+
+    fun makeSadHelloWorldCall() {
```

Comment:
This works!

What I was having in mind was to perhaps modify the other Tab. `SecureStor` and add a button there `Purge token from Runtime`, `Restore token to Runtime`, `Purge token from Disk` etc.

Just to the QA(and us/other devs) has visibility of why the network request fails.

Reply:
mmmm true, think that functionality would be nice but maybe too much to add to this 'ticket'